phlib/fastlock.c

```diff
@@ -24,15 +24,15 @@
 #define PH_LOCK_EXCLUSIVE_WAKING 0x2
 
 #define PH_LOCK_SHARED_OWNERS_SHIFT 2
-#define PH_LOCK_SHARED_OWNERS_MASK 0x3ff
+#define PH_LOCK_SHARED_OWNERS_MASK 0x3fful
```

Comment:
These types haven't changed in 14 years. This will change some of the lock expressions from signed to unsigned and I'm not sure if there will be side effects?

Reply:
The problem manifests here: https://github.com/winsiderss/systeminformer/blob/882e8adcb930e493a30d449d526679730dce226b/phlib/fastlock.c#L40
A shift that shifts into sign bit. C++20 changed it to IB, but we're doing C here.

The actual code usages work on ULONGs, so the issue doesn't happen there
https://github.com/winsiderss/systeminformer/blob/882e8adcb930e493a30d449d526679730dce226b/phlib/fastlock.c#L232-L238